### ISSUE TITLE
headTags are ignored

### ISSUE THREAD

**Issue by rastislavcore** [facebook/docusaurus#9792]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#issues).

### Prerequisites

- [X] I'm using the latest version of Docusaurus.
- [X] I have tried the `npm run clear` or `yarn clear` command.
- [X] I have tried `rm -rf node_modules yarn.lock package-lock.json` and re-installing packages.
- [X] I have tried creating a repro with https://new.docusaurus.io.
- [X] I have read the console error message carefully (if applicable).

### Description

headTags are not reflected in builds. Defined in: https://github.com/CorePass/corepass-docs/blob/4695736a717cc14a3ac8154cf34fdf198a100d30/docusaurus.config.ts#L136

Tried one example from https://docusaurus.io/docs/seo#global-metadata

### Reproducible demo

https://docs.corepass.net

### Steps to reproduce

1. Create the item in headTags
2. Setup the parameters
3. Build or publish the website

### Expected behavior

Head elements should be shown and generator should be replaced.

### Actual behavior

No head tags are injected. It is ignored.

### Your environment

- Public source code: https://github.com/CorePass/corepass-docs/tree/master
- Public site URL: https://docs.corepass.net
- Docusaurus version used: 3.1.0
- Environment name and version (e.g. Chrome 89, Node.js 16.4): Chromium Engine Version 120.0.6099.199; Node v20.7.0
- Operating system and version (e.g. Ubuntu 20.04.2 LTS): macOS 13.5.2

### Self-service

- [ ] I'd be willing to fix this bug myself.

**Comment by swaroopar** [comment]
+1 have exactly same issue with version 3.1.0

**Comment by Josh-Cena** [comment]
Hi, `headTags` is a top-level config, not a `themeConfig`. @swaroopar please confirm if yours is the same issue.

**Comment by Josh-Cena** [comment]
Yeah looks like that's wrong. I'll see if there are other places.

**Comment by g0rbe** [comment]
[The search for `headTags`](https://github.com/search?q=repo%3Afacebook%2Fdocusaurus+headTags) shows only this page wrong.
